src/activity_handlers.cpp

```diff
@@ -460,20 +463,19 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
         }
     }
 
-    if( pieces <= 0 ) {
-        p->add_msg_if_player(m_bad, _("Your clumsy butchering destroys the flesh!"));
-    } else {
-        p->add_msg_if_player(m_good, _("You harvest some flesh."));
-        const itype_id meat = corpse->get_meat_itype();
-        if( meat == "null" ) {
-            return;
-        }
+    //now handle the meat, if there is any
+    //bone_tainted is handled above (treated as a byproduct, not a meat)
+    if( pieces <= 0 && meat != "null" && meat != "bone_tainted" ) {
```

Comment:
> Please try to avoid hardcoded ids wherever possible.

Noted, thanks.

> Also, the part of condition `meat != "null" && meat != "bone_tainted"` occurs in two adjacent branches (which is not good).

How so?